### ISSUE TITLE
[Question/Feature Request] Trigger Matching Branches

### ISSUE THREAD

**Issue by s0undt3ch** [jenkinsci/ghprb-plugin#156]
We have a setup where every pull request get's it's own set of jobs including a "master build" for that PR which receives the GHPRB triggers.

Since multiple jobs are "listening" to the triggers, when a PR is updated/synchronized, all "master build" which listed to the triggers get built.

Is there a way to "choose" the right "master build" by the PR branch name(`pr/<pr-number>/head`) or refspec (`+refs/pull/*:refs/remotes/origin/pr/<pr-number>`)?

If not, could this be added?

**Comment by DavidTanner** [comment]
You would do this by using the trigger phrase, and then only building with the trigger

**Comment by s0undt3ch** [comment]
That's not something the plugins exposes is it? Ie, it would have to be implemented?

**Comment by DavidTanner** [comment]
There are two settings on the project page for the trigger.  They are currently exposed.  The trigger phrase is an exact match and didn't use a regex like the test or whitelist phrases.

**Comment by s0undt3ch** [comment]
But that matches against a PR comment right?

**Comment by s0undt3ch** [comment]
I created a flow job which gets all PR triggers which then triggers the exact PR jobs(while a feature to support this is not implemented).

**Comment by DavidTanner** [comment]
I am guessing I am not understanding what you are looking for then.  How would you decide where the job goes?  Are you looking for something in the PR description that the trigger would key off of whether to build or not?

**Comment by s0undt3ch** [comment]
The ghprb incoming payload includes the pull request id, among other information. My idea was to allow matching job names `jobs/foo/bar/{ghPrId}/the-build`(this would mean searching for all jobs)...

**Comment by DavidTanner** [comment]
Maybe a better way would be to put a new extension in the trigger that limits the job to triggering certain PR's only?  The problem here is that you have to create a new job per pull request right?

**Comment by s0undt3ch** [comment]
Yes, I create a "set" of jobs per pull request, see an example [here](https://jenkins-testing.saltstack.com/job/bootstrap/job/pr/). Each of those jobs has a `main-build` which should get triggered, but only for the matching PR.

**Comment by DavidTanner** [comment]
Sounds like a whitelist for PR numbers.

**Comment by s0undt3ch** [comment]
But that needs to be able to be set on each of the necessary jobs(not just globally)...

**Comment by s0undt3ch** [comment]
Is the "Whitelist Target Branches" the implementation of this?

**Comment by s0undt3ch** [comment]
Hmm, don't think so, what I need is to white-list the source, not the target....

**Comment by s0undt3ch** [comment]
Have gave any thoughts on implementing this?

**Comment by benpatterson** [comment]
@s0undt3ch I'm helping out with this repo. I do think the whitelisting of branches is the feature you're looking for. If this issue is still relevant to you, could you give that a try?

**Comment by s0undt3ch** [comment]
I'm looking at white listing yes, but the source branch, not the destination branch which was what was available last time I looked.

Can we white list the source branch now?

**Comment by benpatterson** [comment]
Ah, I don't think so. Marking this as an enhancement (but I need to doublecheck, too).

**Comment by gjabouley-invn** [comment]
Hello
any update on the "Source Branch whitelist" feature?
I could dedicate some time to implement it, do you know if there is ongoing work on it?
